test/features/parsoid/transform.js

```diff
@@ -15,197 +16,166 @@ describe('transform api', function() {
+            { html: '<h2></h2>', scrub_wikitext: 1 }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
-            assert.deepEqual(res.body, '');
+            assert.deepEqual(res.text, '');
         });
     });
 
     it('supports reversed order of properties in TimeUuid meta', () => {
         const newHtml = testPage.html.replace(/<meta property="mw:TimeUuid" content="([^"]+)"\/?>/,
             '<meta content="$1" property="mw:TimeUuid" />');
-        return preq.post({
-            uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
-            body: {
-                html: newHtml
-            }
-        })
+        return client.post(
+            `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
+            { html: newHtml }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
             const pattern = /Selser test/;
-            if (!pattern.test(res.body)) {
+            if (!pattern.test(res.text)) {
                 throw new Error(`Expected pattern in response: ${pattern
                 }\nSaw: ${JSON.stringify(res, null, 2)}`);
             }
-            assert.contentType(res, contentTypes.wikitext);
+            assert.match(res.headers['content-type'], mwUtil.constructRegex([contentTypes.wikitext]));
         });
     });
 
     it('supports stashing content', () => {
-        return preq.post({
-            uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/wikitext/to/html/${testPage.title}/${testPage.revision}`,
-            body: {
-                wikitext: '== ABCDEF ==',
-                stash: true
-            }
-        })
+        return client.post(
+            `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/wikitext/to/html/${testPage.title}/${testPage.revision}`,
+            { wikitext: '== ABCDEF ==', stash: true }
+        )
         .then((res) => {
             assert.deepEqual(res.status, 200);
             const etag = res.headers.etag;
             assert.deepEqual(/\/stash"$/.test(etag), true);
-            return preq.post({
-                uri: `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
-                headers: {
-                    'if-match': etag
-                },
-                body: {
-                    html: res.body.replace('>ABCDEF<', '>FECDBA<')
-                }
-            });
+            return client.post(
+                `${server.config.baseURL('en.wikipedia.beta.wmflabs.org')}/transform/html/to/wikitext/${testPage.title}/${testPage.revision}`,
+                { 'if-match': etag, html: res.text.replace('>ABCDEF<', '>FECDBA<') }
```

Comment:
Yeah, ofc it's failing, 'if-match' is a header, not a part of the body. 

Unfortunately it seems like there's no way to set custom headers in api-testing, lemme chat with @clarakosi and decide how to add that capability.

Reply:
I've updated the HTTP methods to now accept an object of headers as the third parameter. Please update this to pass the `if-match` headers

https://github.com/wikimedia/mediawiki-tools-api-testing/blob/master/lib/REST.js#L103

Reply:
Thanks @clarakosi and @Pchelolo, the `if-match` header has been added and the `supports stashing content` test is now passing.